nixos/modules/security/acme/default.nix

```diff
@@ -1177,15 +1180,26 @@ in
               roundRobinApplyAttrs selfsignServiceFunctions concurrencyLockfiles
             else
               lib.mapAttrs (_: f: f null) selfsignServiceFunctions;
+          # This indirection is needed because the actual renewal units remain
+          # active so that s-t-c can update them based on changes.
+          triggerRenewServices = lib.mapAttrs' (
```

Comment:
It might make sense to split up  into two services.  One for `lego run` (request new certificate) and one for `lego renew`.

`lego renew` only runs through timer. `lego run` can instead by pulled in by `multi-user.target` (or whatever service pulls it in).

Then I think the stuff here gets a bit clearer